### ISSUE TITLE
Default route is sometimes erroneously removed

### ISSUE THREAD

**Issue by lboclboc** [NetworkConfiguration/dhcpcd#136]
Sometimes we see that the default route that lies on eth0 is removed when the link of a local network using usb0 goes down. We are using version 9.4.1. on raspberry pi. See log below:

Oct  3 11:20:13 raspberry-01 dhcpcd5[400]: usb0: carrier lost
Oct  3 11:20:13 raspberry-01 kernel: [ 1885.303189] usb 1-1.2: USB disconnect, device number 7
Oct  3 11:20:13 raspberry-01 kernel: [ 1885.303475] rndis_host 1-1.2:1.0 usb0: unregister 'rndis_host' usb-3f980000.usb-1.2, RNDIS device
Oct  3 11:20:13 raspberry-01 dhcpcd5[400]: usb0: deleting route to 169.254.1.0/24
Oct  3 11:20:13 raspberry-01 dhcpcd5[400]: eth0: deleting route to 10.46.29.0/24             <- I think this is wrong
Oct  3 11:20:13 raspberry-01 dhcpcd5[400]: eth0: deleting default route via 10.46.29.1    <- and this
Oct  3 11:20:13 raspberry-01 dhcpcd5[400]: usb0: removing interface

**Comment by rsmarples** [comment]
That does look wrong.

**Comment by lboclboc** [comment]
Yes, I forgot to mention that we also have a number of DNAT iptables rules (static ones added at boot). If that somehow can cause problems. Not for port 67 or 68 though.

**Comment by perkelix** [comment]
Does the above still apply to 10.5.2?

**Comment by lboclboc** [comment]
Sorry, we worked around it and used another approach so I cant really test it now
